apis/nucleus/src/components/listbox/components/ListBoxSearch.jsx

```diff
@@ -75,27 +76,30 @@ export default function ListBoxSearch({
   const { translator } = useContext(InstanceContext);
   const [value, setValue] = useState('');
   const [wildcardOn, setWildcardOn] = useState(false);
-
   const inputRef = useRef();
 
   const { getStoreValue, setStoreValue } = useDataStore(model);
   const isRtl = direction === 'rtl';
 
   const cancel = () => selections.isActive() && selections.cancel();
 
-  const abortSearch = async () => {
-    // When select is disabled we always want to allow abort (but not permitting selections).
-    const preventAbort = selectionState.selectDisabled() ? false : !selections.isModal();
-    if (preventAbort) {
-      return;
-    }
+  const abortListObjectSearch = async () => {
     try {
       await model.abortListObjectSearch(TREE_PATH);
     } finally {
       setValue('');
     }
   };
 
+  const abortSearch = () => {
+    // When select is disabled we always want to allow abort (but not permitting selections).
+    const preventAbort = selectionState.selectDisabled() ? false : !selections.isModal();
+    if (preventAbort) {
+      return;
+    }
+    abortListObjectSearch();
+  };
```

Comment:
I don't think this should be split into two functions

If we want to totally disconnect seach and selection then preventAbort should be removed in all cases
but if that was not the intention from the change in https://github.com/qlik-oss/nebula.js/pull/1531
then I think we should make start typing in the search input also enter selection mode and then I think the preventAbort should be OK to keep for all cases

Reply:
Understand. I'm also a bit confused that what I should expect from filterpane in this case! I think not many really paid attention how the new changes on table searchbox might affect the filterpane, neither did I. It's good you brought it up then we can fix it.
I'll ask and choose one of your suggestions base on that! I'll let you know 🙏

Reply:
I know too little on how Listbox works but I have some similar/related issues so I tested a little and it seems like this [one](https://github.com/qlik-oss/nebula.js/blob/7560af739b36f3c28ebf6afffd7adc198414b783/apis/nucleus/src/components/listbox/components/ListBoxSearch.jsx#L84) trigger the change of `visible` and the search is aborted.

What i did is to change the code to this  `const cancel = () => selections.cancel();` not sure if cancelling even though  the selection haven't began will cause side effect.

Reply:
@T-Wizard I asked UX (Patrik L.) about this in filterpane. He was agree with the test results I showed in the video. We want the searchbox to be aborted regardless there is any selection or not. Pressing escape key should cancel the unconfirmed selections and hide searchbox in filterpane and close the popover listbox in table.
@linhnguyen-qlik I probably couldn't recognize what you referred to...what is wrong with aborting search in your case? don't you expect it to be aborted ?

Reply:
> @linhnguyen-qlik I probably couldn't recognize what you referred to...what is wrong with aborting search in your case? don't you expect it to be aborted ?

No there is nothing wrong with abortSearch it is just that with the code I mentioned above you could have a fix with just one line of code.

Reply:
Maybe we are talking about different fixes ? 
The purpose of this PR is to be sure that Esc key cancels the selections if there is any, returns the focus back to the initial listbox and be sure the searchbox is gone. It's slightly different with table searchbox which I cancel the selections and close the popover listbox immediately because there we don't need it to stay open, this is the reason why I propagate the event at some points to close whole the listbox. Do these steps break anything from cyclic perspective ?

Reply:
I tried Linh's solution and it sounds working with the known scenarios. From now and then we cancel all selections regardless they are active or not and the searchbox will be closed when Esc key is pressed. I think we don't need preventAbort anymore since we don't want to stop closing searchbox anyway. 

@T-Wizard & @linhnguyen-qlik I don't know about the side effects but we know that the changes have passed at least the existing tests. If you are agree, maybe you want to approve this PR? :)

Reply:
Maybe you could revert/remove the code that are no longer relevant?